src/metro/devices/display/fast_plot.py

```diff
@@ -1235,8 +1235,11 @@ def dataSet(self, d):
     def dataAdded(self, d):
         self.ch_data = d
 
+        nan_replacement = 0
         if isinstance(self.ch_data, xr.DataArray):
             self.idx_data = self.ch_data.data[self.index]
+            if "replace_nan" in self.ch_data.attrs:
+                nan_replacement = self.ch_data.attrs["replace_nan"]
```

Comment:
One could also allow specifying and utilizing the various `dropna` / `fillna` etc. available for `DataArray`. They don't handle `inf`, though. Also, dropping is tricky.